contracts/base/escrow/dapps/Compound.sol

```diff
@@ -38,87 +39,94 @@ contract Compound is ICompound, BaseEscrowDapp {
     // State is shared with Escrow contract as it uses delegateCall() to interact with this contract.
 
     /**
-        @notice To lend we first have to approve the cToken to access the token balance then mint. 
-        @param cTokenAddress address of the token.
-        @param amount amount of tokens to mint. 
+        @notice To lend we first have to approve the cToken to access the token balance then mint.
+        @param tokenAddress address of the token.
+        @param amount amount of tokens to mint.
     */
-    function lend(address cTokenAddress, uint256 amount) public onlyOwner() {
-        require(cTokenAddress.isContract(), "CTOKEN_ADDRESS_MUST_BE_CONTRACT");
-        CErc20Interface cToken = CErc20Interface(cTokenAddress);
-        uint256 balanceBeforeMint = cToken.balanceOf(address(this));
-        IERC20 underlying = IERC20(cToken.underlying());
-        require(
-            underlying.balanceOf(address(this)) >= amount,
-            "COMPOUND_INSUFFICIENT_UNDERLYING"
-        );
+    function lend(address tokenAddress, uint256 amount) public onlyOwner() {
+        require(_balanceOf(tokenAddress) >= amount, "COMPOUND_INSUFFICIENT_UNDERLYING");
 
-        underlying.approve(cTokenAddress, amount);
+        CErc20Interface cToken = _getCToken(tokenAddress);
+        uint256 balanceBeforeMint = cToken.balanceOf(address(this));
+        IERC20(tokenAddress).approve(address(cToken), amount);
```

Comment:
I think we need to validate the returned value

Reply:
If it is not successful, then it will fail on the next line when we try to mint.